Model/Form/Modifier/AddressFormModifiers.php

```diff
@@ -103,7 +105,10 @@ public function applyHideVatIdFieldForMerchantCountry(EntityFormInterface $form)
             return;
         }
 
-        if ($countryField->getValue() === $this->configuration->getMerchantCountryCode()) {
+        $isVatIdFieldVisible = $countryField->getValue() === $this->euvatConfiguration->getMerchantCountryCode() &&
```

Comment:
I think in this case variable name might be confusing a bit.
 Probably smth like $isVatIdHidden would make more sense here?
 It's a bit confusing, that we hide vat_id when $isVatIdFieldVisible == true

Reply:
Nice spot!